png_cicp_chunk_proposal.md

```diff
@@ -0,0 +1,50 @@
+# Adding support for HDR imagery to the PNG format
+Authors (alphabetical): Chris Blume, Chris Seeger, Pierre-Anthony Lemieux
+
+Status: Draft
+
+## Problem to be solved
+The gAMA chunk of the Portable Network Graphics (PNG) format (specified in [PNG]) parameterized the transfer function of the image as a power law. As such, it cannot model the Reference PQ or HLG OTFs specified in [BT2100-1], which are commonly used for HDR images.
+
+This specification uses the existing iCCP chunk to unambiguously signal the color system of an image that uses the Reference PQ EOTF specified in [BT2100-1]. It also allows graceful processing by decoders that do not conform to this specification by recommending fallback values for the gAMA chunk, cHRM chunk, and embedded ICC profile.
+
+An existing W3C group note (https://www.w3.org/TR/png-hdr-pq/) specifies an approach which is limited: it supports signaling of only the ITU BT.2100 PQ EOTF and uses magic values in the iCCP chunk to signal color spaces.
+
+[H.273](https://www.itu.int/rec/T-REC-H.273/en) specifies a controlled vocabulary for the parameterization of
+color space information:
+* Color Primaries
+* Transfer Function
+* Matrix Coefficients (only required if essence is stored as Y’CbCr)
+
+Use [ITU-T Series H Supplement 19](https://www.itu.int/rec/T-REC-H.Sup19-201910-I) “recommendations” to signal the controlled vocabulary for image formats commonly used for baseband linear broadcasts or file-based Video-on-Demand(VOD) services.
+* Tables within this document summarize values most commonly used in broadcast for 47 different standards documents including H.273
+
+
+## Basic requirements
+* does not break current implementations
+* extensible signaling of color space based on H.273
+* does not require the presence of iCCP chunk and embedded ICC profiles
+* cICP chunk comes before frame data
+
+## Strawman approach
+Define a cICP chunk that contains the 7 bytes necessary to carry the
```

Comment:
`cICP` would not be a valid chunk name (as per PNG spec, 5.4), specifically the four character since I would consider it "safe to copy" which means it should be lowercase - thus `cICp`

Reply:
It is not clear to me whether this is intended to be safe to copy or not. But  iCCP  is also not safe to copy.

Reply:
Nor indeed is `gAMA` and `cHRM`

Reply:
Why would you consider the profile and gamma info not safe to copy?  They are absolutely safe to copy since the assumption is that you aren't changing the colorspace of the image data when processing it.

Reply:
Looking at http://w3.org/TR/PNG-Structure.html#Chunk-naming-conventions I think I understand why those chunks are unsafe to copy and why this new chunk should be as well.

"If the program has made _any_ changs to _critical_ chunks, ... then unrecognized unsafe chunks must **not** be copied to the output PNG file. (Of course, if the program **does** recognize the chunk, it can choose to output an appropriately modified version.)"

So for example, if the program doesn't understand the HDR chunk and modifies some pixel values, it might do so in the wrong color space. I don't think we want to preserve the incorrect color space. I think we want to throw out the HDR information and now have a file in sRGB, for example.